includes/class-wc-gateway-ppec-checkout-handler.php

```diff
@@ -114,11 +114,13 @@ function checkout_init( $checkout ) {
 	 * @return array
 	 */
 	public function filter_default_address_fields( $fields ) {
+		$require_phone_number = wc_gateway_ppec()->settings->require_phone_number;
+
 		if ( method_exists( WC()->cart, 'needs_shipping' ) && ! WC()->cart->needs_shipping() ) {
 			$not_required_fields = array( 'address_1', 'city', 'state', 'postcode', 'country' );
 			foreach ( $not_required_fields as $not_required_field ) {
 				if ( array_key_exists( $not_required_field, $fields ) ) {
-					$fields[ $not_required_field ]['required'] = false;
+					$fields[ $not_required_field ]['required'] = 'no' !== $require_phone_number;
```

Comment:
Here's how to test it:
 .
* Start checkout from cart page.
* When placing order, you'll get:

   <img width="832" alt="screen shot 2017-06-07 at 1 04 29 am" src="https://user-images.githubusercontent.com/78313/26844357-55104522-4b1d-11e7-8251-26031bf81eb5.png">

Whether **Require Phone Number** is enabled or not, you'll get the same notice.